cpp/src/arrow/meson.build

```
@@ -289,6 +370,63 @@ install_headers(
     install_dir: 'arrow',
 )
 
+if needs_tests
+    filesystem_dep = dependency(
+        'boost',
+        modules: ['filesystem'],
```

Do we need `system` too?

```suggestion
        modules: ['filesystem', 'system'],
```

---

I don't believe we need that at the moment. If we do, I think we might need to also have `filesystem_dep` and `system_dep` be separate dependencies rather than included in one (unless we don't ever care to differentiate between them)